kratos/utilities/particles_utilities.h

```diff
@@ -0,0 +1,330 @@
+                if(is_found){
+
+                    auto& r_geom = p_element->GetGeometry();
+                    for(unsigned int i=0; i<r_geom.size(); ++i){
+                        auto& rcounter = r_geom[i].GetValue(rCounterVariable);
+                        AtomicAdd(rcounter, 1.0);
+                    }
+                }
+
+            });
+
+        }
+
+    /**
+    this function looks if particle is found in the locator. If it is not, rVariable is marked with the value "outsider_value"
+    */
+    template<unsigned int TDim, class TScalarType>
+    static void MarkOutsiders(
+            BinBasedFastPointLocator<TDim>& rLocator,
+            ModelPart& rParticlesModelPart,
+            const Variable<TScalarType>& rVariable,
+            const TScalarType outsider_value
+            )
+        {
+            unsigned int max_results = 10000; 
+            typename BinBasedFastPointLocator<TDim>::ResultContainerType TLS(max_results);
+            
+            //for every interface node (nodes in cut elements)
+            block_for_each(rParticlesModelPart.Nodes(), TLS, [&rLocator, &rVariable, outsider_value](auto& rNode, auto& rTLS){
+
+                Vector shape_functions;
+                Element::Pointer p_element;
+                bool is_found = rLocator.FindPointOnMesh(rNode.Coordinates(), shape_functions, p_element, rTLS.begin(), rTLS.size(), 1e-5);
+
+                if(!is_found){
+                    rNode.SetValue(rVariable, outsider_value);
+                }
+            });
+        }
+
+    /**@brief provides the value of the interpolated variable "rInterpolationVariable" at all the positions indicated in rCoordinates
+    */
+    template<unsigned int TDim, class TScalarType>
+    static std::pair< std::vector<bool>, std::vector<TScalarType> > InterpolateValuesAtPoints(
+            BinBasedFastPointLocator<TDim>& rLocator,
+            const Matrix& rCoordinates,
+            const Variable<TScalarType>& rInterpolationVariable
+            )
+        {
+            unsigned int max_results = 10000; 
+            typename BinBasedFastPointLocator<TDim>::ResultContainerType TLS(max_results);
+
+            auto interpolations = std::make_pair(std::vector<bool>(rCoordinates.size1()), std::vector<TScalarType>(rCoordinates.size1()));            
+            
+            //for every interface node (nodes in cut elements)
+            IndexPartition(rCoordinates.size1()).for_each(TLS, [&rLocator, &rCoordinates, &interpolations, &rInterpolationVariable](const auto& i, auto& rTLS){
+
+                Vector shape_functions;
+                Element::Pointer p_element;
+                bool is_found = rLocator.FindPointOnMesh(row(rCoordinates,i), shape_functions, p_element, rTLS.begin(), rTLS.size(), 1e-5);
+
+                (interpolations.first)[i] = is_found;
+                if(is_found){
+
+                    auto& r_geom = p_element->GetGeometry();
+                    (interpolations.second)[i] = TScalarType();
```

Comment:
```suggestion
                    (interpolations.second)[i] = rInterpolationVariable.Zero();
```
I'm not sure if all the data types default constructors return their corresponding zero.

Reply:
why not TScalarType()? it should be the same here ... and i suspect that it can be faster since i am not accessing million of times to the same variable in the memory to get the zero

Reply:
My though was in the direction of extending it to non scalar types. In these cases I'm not sure what the default constructor of the type does.

Reply:
ok, done